### DISCUSSION TITLE
How to use resolver in the async context

### DISCUSSION THREAD

**Discussion by ghost** [RGB-WG/rgb#220] [Q&A]
https://github.com/RGB-WG/rgb/blob/master/src/resolvers/any.rs#L44

```
pub struct AnyResolver {
    inner: Box<dyn RgbResolver>, // here missing Send
    terminal_txes: HashMap<Txid, Tx>,
}
```

**Comment by dr-orlovsky** [comment]
It is a feature, not a bug :)

If you need one for async, you can write your own and name it AnyResolverAsync (+corresponding Electrum and Esplora imlementations)

**Comment by dr-orlovsky** [comment]
The recommended way of doing non-blocking resolver is to create a dedicated thread which permanently run resolver and talk to it via `crossbeam`. Than, you need to write `ResolverProxy` which will be on the other end of the crossbeam channel and will send commands to the resolver thread - and be used inside rb-wallet and rgb APIs. In this way you do not need async in RGB itself and have a non-blocking thing
